lib/util/viewer.js

```diff
@@ -406,6 +387,30 @@ export function getColorAndNameFromRoute(operator, route) {
   }
 }
 
+/**
+ * Helper method to determine a route color based on config color settings.
+ *
+ * Colors can be one of the following configured values:
+ * - gtfs (default): colors are taken directly from gtfs/operator config
+ * - gtfs-softened: colors are taken from gtfs/operator config and "softened" to be less intense
+ * - disabled: color is always a generic gray
+ */
+export function getRouteColorBasedOnSettings(operator = {}, route = {}) {
```

Comment:
I would much rather get-a-`RouteNameRenderer`-based-on-settings, and each route renderer can decide which colors/shapes/etc to use based on the mode and other route attributes.

Reply:
I'm not sure if I understand what you mean. This is what's happening now. The reason that some of the syntax of this is so strange is precisely because the custom route renderer can generate any HTML it would like.